pybamm/expression_tree/concatenations.py

```diff
@@ -179,23 +176,29 @@ def evaluate(self, t=None, y=None):
         return vector
 
 
-def piecewise_constant(neg_value, sep_value, pos_value):
-    """Concatenate three values that don't have a domain into a piecewise constant
-    concatenation. This is useful when we don't want to assign a domain to the inputs
+class PiecewiseConstant(Concatenation):
```

Comment:
Why do we need this class? Why wouldn't the children have domains?

Reply:
One example is the porosities `epsn`, `epss`, `epsp` and reaction rates `sn` and `sp` - we don't want them to have domains so that we can [add them together in the LOQS](https://github.com/tinosulzer/PyBaMM/blob/9dd06cc356d2f495b188ef9a81c1b3ecb5e7f830/pybamm/models/lead_acid/loqs.py#L65) without throwing domain errors, but we do want to plot the porosity as a function of x (in [this line](https://github.com/tinosulzer/PyBaMM/blob/9dd06cc356d2f495b188ef9a81c1b3ecb5e7f830/pybamm/models/lead_acid/loqs.py#L85) we would replace`Concatenation` by `PiecewiseConstant`) or have a piecewise constant reaction rate in the submodels (e.g. replace [this line](https://github.com/tinosulzer/PyBaMM/blob/7a1e87f3e5c04b91e016509b1839d86cbd6ef12c/pybamm/models/submodels/electrolyte_concentration.py#L76) by `PiecewiseConstant(sn, 0, sp)`. 

There is definitely some work to do to make Broadcasts and Concatenations clearer though so I am open to superior alternatives :)

Reply:
ok, lets discuss at the Thurs meeting then. Perhaps things like `epsn`, `epss`, `epsp` can be domain-free, and if a model wants to create a piecewise constant vector over a given set of domains it could write something like:

```python
eps = pybamm.Concatenation(pybamm.Broadcast(epsn,"domain 1"), 
                           pybamm.Broadcast(epss,"domain 2"), 
                           pybamm.Broadcast(epsp,"domain 3"))
```

Reply:
Could we merge this issue and discuss concatenations in the context of #163 ? Just so that we can start using the function-processing part of this issue